model/src/main/java/uk/gov/hmcts/fees2/register/data/service/impl/FeeServiceImpl.java

```diff
@@ -81,6 +84,10 @@
     public Fee save(Fee fee) {
         feeValidator.validateAndDefaultNewFee(fee);
 
+        if (feeValidator.isFeeExists(fee)) {
```

Comment:
it'll give a problem for concurrent requests. May be method synchronized / locks etc?

Reply:
I suppose even if we make method as synchronized it is still not going be concurrent proof, as the API runs in prod runs on multiple nodes.

If we still want to make it synchronized, ideally we should compromise with the performance.